### ISSUE TITLE
Fix Rspec tests

### ISSUE THREAD

**Issue by orangewolf** [scientist-softserv/palni-palci#263]
#Continuation of #509 (Test the Test Suite -- locally && in CI)

Done is when all tests pass in CI

**Comment by DiemBTran** [comment]
Lea Ann - Sept 2022:

---

<details>
./spec/features/create_etd_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/features/collection_editor_role_spec.rb ./spec/requests/admin_dashboard_spec.rb ./spec/features/appearance_theme_spec.rb ./spec/features/appearance_theme_spec.rb ./spec/features/group_spec.rb ./spec/features/admin_set_form_participants_tab_spec.rb ./spec/features/admin_set_form_participants_tab_spec.rb ./spec/features/admin_set_form_participants_tab_spec.rb ./spec/features/admin_set_form_participants_tab_spec.rb ./spec/features/admin_set_form_participants_tab_spec.rb ./spec/features/admin_set_form_participants_tab_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb ./spec/features/collection_reader_role_spec.rb 
./spec/features/collection_reader_role_spec.rb ./spec/jobs/embargo_auto_expiry_job_spec.rb ./spec/jobs/embargo_auto_expiry_job_spec.rb ./spec/jobs/embargo_auto_expiry_job_spec.rb ./spec/jobs/embargo_auto_expiry_job_spec.rb ./spec/features/groups_spec.rb ./spec/features/create_generic_work_spec.rb ./spec/forms/hyrax/forms/permission_template_form_spec.rb ./spec/forms/hyrax/forms/permission_template_form_spec.rb ./spec/forms/hyrax/forms/permission_template_form_spec.rb ./spec/features/cultural_repository_theme_spec.rb ./spec/lib/active_job/queue_adapters/better_active_elastic_job_adapter_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/collection_manager_role_spec.rb ./spec/features/user_roles_spec.rb ./spec/features/user_roles_spec.rb ./spec/features/user_roles_spec.rb ./spec/features/oai_pmh_spec.rb ./spec/features/oai_pmh_spec.rb ./spec/features/oai_pmh_spec.rb ./spec/features/feature_flag_spec.rb ./spec/features/create_oer_spec.rb
</details>

**Comment by DiemBTran** [comment]
Lea Ann - Oct 2022:

---

```
2) Admin Dashboard as an admin user I can hit every url corresponding to each link in the admin dashboard gets the url for activity summary
     Failure/Error: get '/dashboard'
     ActionView::Template::Error:
       no implicit conversion of nil into String
     # /usr/local/bundle/gems/hyrax-3.4.2/app/services/hyrax/analytics/google.rb:80:in `exist?'
     # /usr/local/bundle/gems/hyrax-3.4.2/app/services/hyrax/analytics/google.rb:80:in `auth_client'
     # /usr/local/bundle/gems/hyrax-3.4.2/app/services/hyrax/analytics/google.rb:70:in `token'
     # /usr/local/bundle/gems/hyrax-3.4.2/app/services/hyrax/analytics/google.rb:93:in `user'
     # /usr/local/bundle/gems/hyrax-3.4.2/app/services/hyrax/analytics/google.rb:100:in `profile'
     # /usr/local/bundle/gems/hyrax-3.4.2/app/services/hyrax/analytics/google.rb:192:in `returning_visits_by_day'
     # /usr/local/bundle/gems/hyrax-3.4.2/app/views/hyrax/dashboard/_user_activity_graph.html
```

Feature specs might be failing due to not having analytics id set

UA-238390451-1

**Comment by labradford** [comment]
closed by #548
